exercises/diffie-hellman/tests/diffie-hellman.rs

```diff
@@ -0,0 +1,58 @@
+extern crate diffie_hellman;
+
+use diffie_hellman::*;
+
+#[test]
+fn test_private_key_in_range_key() {
+    let primes: Vec<u64> = vec![5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 773, 9091,
+                                38321, 6131329];
+    let private_keys: Vec<u64> = primes.iter().map(|x| private_key(*x)).collect();
+
+    for i in 0..primes.len() {
+        assert!(1 < private_keys[i] && private_keys[i] < primes[i]);
+    }
+}
+
+#[test]
+#[ignore]
+fn test_public_key_correct() {
+    let p: u64 = 23;
+    let g: u64 = 5;
+
+    let private_key: u64 = 6;
+    let expected: u64 = 7;
```

Comment:
On https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange I see a quote that says:

> Alice chooses a secret integer a = 4, then sends Bob A = g<sup>a</sup> mod p

It seems like thus the expected result should be 8. Is it in fact 8 or did I misunderstand the page?

Reply:
No, you are right.